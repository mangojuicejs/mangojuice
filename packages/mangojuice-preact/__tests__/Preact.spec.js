import AbstractTests from 'mangojuice-react-core/tests';
import Preact from 'preact';
import Subscribe from 'mangojuice-preact';

const React = {
  createElement: Preact.h,
  Component: Preact.Component
};
const ReactDOM = {
  render: Preact.render,
  unmountComponentAtNode: Preact.render.bind(null, () => {})
};
AbstractTests(React, ReactDOM, Subscribe, 'Preact');
