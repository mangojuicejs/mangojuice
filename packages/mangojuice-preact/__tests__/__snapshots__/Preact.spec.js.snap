// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React interface view renderer: Preact shuold render a stateless view of a block 1`] = `
Array [
  Object {
    "a": "test",
    "nested": ChildCmd {
      "createArgs": Array [
        Object {
          "whoo": 123,
        },
      ],
      "logicClass": [Function],
    },
  },
  Object {
    "isChild": true,
    "whoo": 123,
  },
  Array [
    Object {
      "a": "another",
    },
  ],
  Object {
    "a": "another",
  },
]
`;

exports[`React interface view renderer: Preact shuold subscribe to multiple models and bind multiple messages 1`] = `
Array [
  Object {
    "a": "test",
    "nested": ChildCmd {
      "createArgs": Array [
        Object {
          "whoo": 123,
        },
      ],
      "logicClass": [Function],
    },
  },
  Object {
    "isChild": true,
    "whoo": 123,
  },
  Array [
    Object {
      "a": "another",
    },
  ],
  Object {
    "a": "another",
  },
  Array [
    undefined,
  ],
  Array [
    undefined,
  ],
  Array [
    Object {
      "a": "something",
    },
  ],
  Object {
    "a": "something",
  },
]
`;
